packages/components/src/components/Tooltip/TooltipFloatingUi.tsx

```diff
@@ -0,0 +1,213 @@
+import {
+    useFloating,
+    autoUpdate,
+    offset,
+    flip,
+    shift,
+    useHover,
+    useFocus,
+    useDismiss,
+    useRole,
+    useInteractions,
+    FloatingPortal,
+    useDelayGroup,
+    useDelayGroupContext,
+    useMergeRefs,
+    useTransitionStyles,
+    arrow,
+} from '@floating-ui/react';
+import type { Placement, UseFloatingReturn } from '@floating-ui/react';
+import {
+    useState,
+    useMemo,
+    createContext,
+    useContext,
+    ReactNode,
+    HTMLProps,
+    forwardRef,
+    isValidElement,
+    cloneElement,
+    useRef,
+    RefObject,
+    CSSProperties,
+    MutableRefObject,
+} from 'react';
+
+/**
+ * Based on https://floating-ui.com/docs/tooltip but heavily modified
+ */
+
+type ArrowRef = RefObject<SVGSVGElement>;
+
+interface TooltipOptions {
+    initialOpen?: boolean;
+    placement?: Placement;
+    open?: boolean;
```

Comment:
I would prefer `isVisible` (or just `isOpen`) because it feels to me more clear but I understand it can be my subjective opinion.

Reply:
I fixed it in 3e6b44b2ab but please lets have a discussion about whether we shall use `is` or not for boolean variables and props. 

I have some opinion, but not a strong one, so at the end I don't care, but I like to unify it.

Reply:
Yes, I agree. We might have it in the code improvements session, WDYT?

Reply:
yes

Reply:
https://www.notion.so/satoshilabs/Discuss-boolean-naming-isOpen-vs-isOpened-vs-opened-vs-8f0731e487ee4b54b03d250937b6b569?pvs=4